### DISCUSSION TITLE
Replacing reference path, python

### DISCUSSION THREAD

**Discussion by sanders3d** [Autodesk/maya-usd#3437] [General]
I'm having a hard time finding in the documentation how to replace a reference layer with a new path (with python). Any pointers would be helpful.

regards
stefan

**Comment by fkfhvvjn** [comment]
In Python, you can use both OS and OS. exe. The path module handles file and path-related issues. If you want to find and replace new paths to references in a document, follow these steps: use OS. exe. Functions in the path module to parse and manipulate paths. For example, you could use an OS. Path. Join () to join paths, OS. Path. Abspath () to get the absolute path, OS. Path. Dirname () to get the directory where the files are, and so on. Open the document and read the contents. You can use the open () function to open the document and the read () method to read the contents of the document. Use regular expression to find referenced paths. You can use the RE module built into Python to work with regular expression. For example, you can use Re. The search () function searches the document for a specific string or pattern. Replaces the path of the reference. Once you have found the referenced path, you can use the replace () method of the string to replace it with the new path you want. Writes the replacement back to the document. You can open the document again using the open () function and write the replacement back to the document using the write () method.
